ai21/models/chat/function_tool_definition.py

```diff
@@ -0,0 +1,10 @@
+from typing import Optional
+
+from ai21.models.ai21_base_model import AI21BaseModel
+from ai21.models.chat.tool_parameters import ToolParameters
+
+
+class FunctionToolDefinition(AI21BaseModel):
+    name: str
+    description: Optional[str] = None
+    parameters: Optional[ToolParameters] = None
```

Comment:
I would default non Optional parameters to NOT_GIVEN?

Reply:
How about something else without `NotGiven`? Make all of these new classes `TypedDict`, then once a parameter is not passed, it just doesn't pass it (same logic as `NotGiven`. Something like this - 

```python
class FunctionToolDefinition(TypedDict, total=False):
    name: Required[str]
    description: str
    parameters: ToolParameters
```
What do you think? @etang-ai21 would appreciate your thought as well

Reply:
Just to point out. Im not saying to remove `NotGiven` usage from the code. `NotGiven` needs to be used for the `create` method but not within the actual object properties